From 74b9f76fc3dadac0532d8db6146872830b66eb5c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Vi=CC=81ctor=20Marti=CC=81nez?= <knoopx@gmail.com>
Date: Thu, 30 Sep 2010 19:14:47 +0200
Subject: [PATCH] to_xml doesn't work in such case: Event.select('title as t').to_xml [#4840 state:resolved]

NilClass.type is no longer defined in Ruby 1.9 and causes ActiveRecord::Base.to_xml to fail with message: undefined method `type' for nil:NilClass
---
 .../active_record/serializers/xml_serializer.rb    |   10 ++++++++--
 1 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/activerecord/lib/active_record/serializers/xml_serializer.rb b/activerecord/lib/active_record/serializers/xml_serializer.rb
index 15abf8b..0746908 100644
--- a/activerecord/lib/active_record/serializers/xml_serializer.rb
+++ b/activerecord/lib/active_record/serializers/xml_serializer.rb
@@ -226,8 +226,14 @@ module ActiveRecord #:nodoc:
 
     class Attribute < ActiveModel::Serializers::Xml::Serializer::Attribute #:nodoc:
       def compute_type
-        type = @serializable.class.serialized_attributes.has_key?(name) ?
-          super : @serializable.class.columns_hash[name].type
+        case
+          when @serializable.class.serialized_attributes.has_key?(name)
+            type = super
+          when @serializable.class.columns_hash.has_key?(name)
+            type = @serializable.class.columns_hash[name].type
+          else
+            type = NilClass
+        end
 
         case type
         when :text
-- 
1.7.3

